system-overview/cryptography.md

```diff
@@ -0,0 +1,115 @@
+is then uploaded to the Ethereum blockchain using one of your Azimuth private
+keys - either the one associated to the wallet which owns the ship, or the one
+which holds the management proxy.
+
+For moons, the private networking key is generated by the parent ship and
+injected into the `%dawn` sequence for the moon, and its public key is stored by
+the parent in its Jael vane. Moon networking keys may be changed by the parent
+ship, but not by the moon itself.
+
+For comets, their 128-bit `@p` name is actually their public key, and the
+"mining" process to generate a comet consists of guessing a private key for
+which the associated public key has a galaxy that has already been pre-assigned
+to assist its peer discovery process. Thus, comets cannot change their
+networking keys - to get a new private networking key, a new comet must be
+generated. A comet utilizes its private networking key stored in Jael to sign an
+unencrypted attestation packet to verify that it is the owner of the associated
+public key. Because of this, it is currently impossible for a ship to initiate
+communication with a comet - the comet must always be the initiator. This means
+that ultimately two comets cannot communicate with one another unless they have
+somehow verified each others' public keys via some other method. This is merely
+a technical limitation imposed by the design of the system, not an intentional
+handicapping of comet abilities. It is expected that this limitation will
+someday be lifted.
+
+### System components
+
+[Ames](@/docs/arvo/ames/ames.md) is Arvo's networking vane. All packets sent by
+Ames are encrypted utilizing a cryptosuite found in `zuse`. The only exception
+to this are comet self-attestation packets utilized to transmit authentication
+of ownership of the private networking key associated to their public key. Ames
+is responsible for encryption, decryption, and authentication of all packets. By
+default, this utilizes AES symmetric key encryption, whose shared private key is
+got by elliptic curve Diffie-Hellman key exchange of the ships' networking keys.
+
+[Jael](@/docs/arvo/jael/jael-api.md) is primarily utilized for the safe storage
+of private networking keys and retrieval of public networking keys utilized by
+Ames. The Jael vane of planets, stars, and galaxies are responsible for
+distributing the public keys of their moons (ultimately via Ames).
+
+`zuse` is the Hoon standard library. It contains cryptographic functions which
+are utilized by Ames. All cryptographic primitives are
+[jetted](@/docs/vere/jetting.md) in Vere with standard vetted implementations of
+cryptographic libraries.
+
+[Vere](@/docs/vere/_index.md) is Urbit's Nock runtime system, written in C.
+Correct implementation of cryptographic libraries is difficult, and so all
+cryptographic functions implemented in Hoon are hinted to the interpreter to run
+vetted cryptographic libraries reviewed by experts instead.
```

Comment:
Don't make it sound like there's known jet mismatches! Our crypto can and should be good, right?